lib/widget/drawer_genre.dart

```diff
@@ -0,0 +1,81 @@
+import 'package:flutter/material.dart';
+import 'package:the_movie_db/views/movies_genre.dart';
+import '../models/genre.dart';
+import '../models/movie.dart';
+import '../utils/general_constants.dart';
+
+class DrawerGenre extends StatelessWidget {
+  final List<Genre> genres;
+  final List<Movie> movies;
+
+  const DrawerGenre({
+    required this.genres,
+    required this.movies,
+    super.key,
+  });
+
+  static const int mediaQueryConst = 230;
+
+  List<Movie> getMoviesByGenre(int genreId) {
+    return movies.where((movie) => movie.genres.contains(genreId)).toList();
+  }
+
+  List<Widget> _getWidget(BuildContext context) {
+    var children = <Widget>[];
+    children.add(const SizedBox(
+      height: 60,
+      child: DrawerHeader(
+        decoration: BoxDecoration(
+          color: Colors.white24,
+        ),
+        child: Center(
+          child: Text(
+            'Genders',
+            style: TextStyle(
+                fontSize: FontConst.fontTitle,
+                fontWeight: FontWeight.bold,
+                color: Colors.white
+            ),
+          ),
+        ),
+      ),
+    ));
+    children.addAll(genres.map(
+      (genre) {
+        return ListTile(
+          title: Text(
+            genre.name,
+            style: const TextStyle(
+                fontSize: FontConst.fontTitle,
+                fontWeight: FontWeight.bold,
+                color: Colors.white),
+          ),
+          onTap: () {
+            List<Movie> moviesGenre = getMoviesByGenre(genre.id);
+            Navigator.push(
+              context,
+              MaterialPageRoute(
+                builder: (context) => MoviesGenre(
+                  genre: genre,
+                  movies: moviesGenre,
```

Comment:
You shouldn't have the movies here. You should fetch them in the `MoviesGenre` screen